### PR TITLE
Fixes #7572 - remove rundeck from core

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by orrabin** [comment]
This PR needs to be tested as @domcleal suggested in https://github.com/theforeman/foreman/pull/1683

**Comment by ohadlevy** [comment]
regardless of this failure, I propose to move the rundeck specifics into a plugin.

**Comment by orrabin** [comment]
DO NOT MERGE until plugin is ready

File: app/controllers/hosts_controller.rb [hunk 1]
```diff
@@ -43,13 +43,7 @@ def index (title = nil)
...
       end
       format.yaml do
-        render :text => if params["rundeck"]
-          result = {}
-          search.includes(included_associations).each{|h| result.update(h.rundeck)}
-          result
-        else
-          search.all(:select => "hosts.name").map(&:name)
-        end.to_yaml
+          search.all(:select => "hosts.name").map(&:name).to_yaml
```
Comment (ohadlevy) [inline app/controllers/hosts_controller.rb]: I wonder if we need any yaml response anymore, most likely this usage case should move to the api?

**Comment by orrabin** [comment]
@ohadlevy I removed the yaml completely

**Comment by orrabin** [comment]
the plugin has been created and can be found here: https://github.com/theforeman/foreman_host_rundeck

File: app/controllers/hosts_controller.rb [hunk 2]
```diff
@@ -64,7 +55,6 @@ def show
         # summary report text
         @report_summary = Report.summarise(@range.days.ago, @host)
       }
-      format.yaml { render :text => params["rundeck"].nil? ? @host.info.to_yaml : @host.rundeck.to_yaml }
```
Comment (dLobatog) [inline app/controllers/hosts_controller.rb:67]: Rundeck YAML and the conditional should go away, but please leave `@host.info.to_yaml` for pre-API compatibility.

Comment (orrabin) [inline app/controllers/hosts_controller.rb:67]: @eLobato fixed this, should `index` also still have a yaml response?

Comment (dLobatog) [inline app/controllers/hosts_controller.rb:67]: I'm not really familiar with it, so I cannot really tell, sorry.

Comment (orrabin) [inline app/controllers/hosts_controller.rb:67]: @ohadlevy you asked to remove the yaml from `index` but if `show` still has a yaml response for compatibility should I keep it in `index` too?

Comment (dLobatog) [inline app/controllers/hosts_controller.rb:67]: Whoops, sorry, I overlooked that. Yes please keep the YAML response in index too.

**Comment by orrabin** [comment]
@eLobato I returned the yaml response to `show` and `index`

**Comment by dLobatog** [comment]
Thanks @orrabin , will merge if Jenkins is :green_apple:

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/b12d9add35094212043a7763c3556374f6348110 , thanks again @orrabin !

I've added a comment to the commit pointing to the plugin, and I've made a mental note to put it in this month's newsletter to warn users.

**Pull request closed without merging**